### DISCUSSION TITLE
Two point correlation output

### DISCUSSION THREAD

**Discussion by PascalRuzzante** [PMEAL/porespy#626] [Q&A] [answered]
I have a quick question regarding the output of the two point correlation (fft) function  -namely that according the Torquato, the two point correlation should be equal to your porosity/phase fraction ᶲ at r = 0, then decay to ᶲ^2 as r goes to inf. However, the function in porespy seems to output something close to 100% probability at r = 0, decaying down the the void fraction ᶲ as r goes to inf. 

My question is then whether the probability output of the function is supposed to be the same as the void fraction at r=0, or if they are two different properties that I am getting mixed up. If they are two separate properties, do you know what the relationship is between them?

Note, that the example code on the website demonstrates what I am talking about. https://porespy.org/examples/metrics/tutorials/two_point_correlation.html?highlight=two%20point 

Thanks in advance,
Pascal

**Comment by jgostick** [comment, marked as answer]
Surprizingly you are the first person to call us out on this.  In our view, the y-axis is the probability that 2 pixels are in the same phase.  So the odds that two pixels are in the same phase should be 1 at r=0, and approach the porosity as r->inf.  Also, approaching y=0 at r->inf, as you suggest, would result in some negative correlation values since the function oscillates around the global porosity, which seems odd to me.  HOWEVER, I concede that we are not experts in the use of this function, so if our interpretation is a problem, we are open to change.  Hopefully I am correct in thinking that there is a direct scaling to get to our data to map onto the range you suggest, like ``(y-phi)*phi/(1-phi)``.

**Reply by PascalRuzzante** [reply]
Thanks for the quick answer! I am by no means an expert on the function either - just beginning a reconstruction project, and learning all of the stuff involved with it. Porespy has been very helpful so far!

You are correct in that the scaling is easy, however one correction is that the range I was suggesting goes from phi(at r=0) to phi^2 as 
r-> inf (in which case, I just have to scale by a factor of phi). I think this might get rid of the negative correlation/oscillating values issue you mention.

On a theoretical level though, my interpretation of the two-point correlation is that the indicator function for the pore phase would only return True if both line endpoints are in the pore phase(in the case of r=0, both end points refer to the same voxel/pixel). I think the function in porespy is returning True if both points are in the pore phase OR both points are in the solid phase. Is that what you intended?

**Reply by jgostick** [reply]
Yes, good point about the phi^2...my answer was a bit hasty.  EDIT: But your second point about returning True for the pore phase OR the void phase is not correct.  The function works by always selecting the first pixel in the void phase, then chooses a second pixel far away.  If they are both in the void phase then this returns True.  This is why the results starts at 1 and ends at phi.  (I had to edit this after I thought back to how the function works).  

I guess this could use some clarification.  Perhaps we should consider either augmenting this function to optionally return the alternative version you suggest, or just add a whole new function.

**Reply by OhmFF** [reply]
Thanks for the discussions.
I have been using the two-point correlation to characterize the pore structure of cementitious materials. Actually, I have the exact same question. After reviewing the literature, the definition seems different in each literature.
For example, https://doi.org/10.1016/j.conbuildmat.2019.01.030 (see section 3.3.3) the definition of two-point correlation is the same as @PascalRuzzante. However, https://doi.org/10.1557/jmr.2010.41 the probability is 1 at r = 0 which is the same as Porespy's. I think it may depend on the interpretation. 

For my intuition, I think the return probability of 1.0 at r = 0 makes sense for a specific phase of interest.
As mentioned in #666, is there a way to cross-validate the results with other analytical functions?

**Reply by PascalRuzzante** [reply]
The second paper you linked is equivalent to porespy's definition - you're correct. However, it looks like they just normalized the two-point correlation by the porosity of each sample (which by Torquato's definition, is equivalent to S2(r=0)). This is probably fine for qualitative analysis. I suppose it depends on what application you are using it for. If you are trying to compare two different materials (such as in simulated annealing reconstruction), particularly if they have different porosities, I would argue it makes sense to use Torquato's definition, so that the difference in porosity is shown. Otherwise, they are both normalized by their own porosity. 

Additionally, relations like 
![image](https://user-images.githubusercontent.com/98714796/161793935-76aabccd-9cd6-4e8a-80b3-155c13083b1c.png)
S. Torquato, Random Heterogeneous Materials, Springer, Princeton, 2002.
for surface area would also have to be scaled by the porosity.
